src/commands/synthetics/run-test.ts

```diff
@@ -177,29 +179,65 @@ export class RunTestCommand extends Command {
   }
 
   private async getTestsList(api: APIHelper) {
-    if (this.testSearchQuery) {
-      const testSearchResults = await api.searchTests(this.testSearchQuery)
+    if (this.config.testSearchQuery) {
+      const testSearchResults = await api.searchTests(this.config.testSearchQuery)
 
       return testSearchResults.tests.map((test) => ({config: this.config.global, id: test.public_id}))
     }
 
-    const listOfGlobs = this.fileGlobs || [this.config.files]
-
-    const suites = (await Promise.all(listOfGlobs.map((glob: string) => getSuites(glob, this.reporter!))))
+    const suites = (await Promise.all(this.config.files.map((glob: string) => getSuites(glob, this.reporter!))))
       .reduce((acc, val) => acc.concat(val), [])
       .map((suite) => suite.tests)
       .filter((suiteTests) => !!suiteTests)
 
     const testsToTrigger = suites
       .reduce((acc, suiteTests) => acc.concat(suiteTests), [])
       .map((test) => ({
-        config: {...this.config!.global, ...test.config},
+        config: {...this.config.global, ...test.config},
         id: test.id,
       }))
 
     return testsToTrigger
   }
 
+  private async resolveConfig() {
+    // Default < file < ENV < CLI
+
+    // Override with file config variables
+    const configPath = this.configPath ?? this.config.configPath
+    this.config = await parseConfigFile(this.config, configPath)
+
+    // Override with ENV variables
+    this.config = deepExtend(
+      this.config,
+      removeUndefinedValues({
+        apiKey: process.env.DATADOG_API_KEY,
+        appKey: process.env.DATADOG_APP_KEY,
+        datadogSite: process.env.DATADOG_SITE,
+        subdomain: process.env.DATADOG_SUBDOMAIN,
+      })
+    )
+
+    // Override with CLI parameters
+    this.config = deepExtend(
+      this.config,
+      removeUndefinedValues({
+        apiKey: this.apiKey,
+        appKey: this.appKey,
+        configPath: this.configPath,
+        files: this.files,
+        publicIds: this.publicIds,
+        testSearchQuery: this.testSearchQuery,
+        tunnel: this.tunnel,
+      })
+    )
+
+    if (typeof this.config.files === 'string') {
+      this.reporter!.log('[DEPRECATED] "files" should not be a string, array of string expected. The conversion will be automatic')
```

Comment:
```suggestion
      this.reporter!.log('[DEPRECATED] "files" should be an array of string instead of a string. The conversion will be automatic')
```